server/api.go

```diff
@@ -645,8 +634,127 @@ func (a *API) getConnectedUsersFile(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	if !a.p.API.HasPermissionTo(userID, model.PermissionManageSystem) {
+		a.p.API.LogWarn("Insufficient permissions", "user_id", userID)
+		http.Error(w, "not able to authorize the user", http.StatusForbidden)
+		return
+	}
+
+	whitelist, err := a.p.getWhitelistEmails()
+	if err != nil {
+		a.p.API.LogWarn("Unable to get connected users list", "error", err.Error())
+		http.Error(w, "unable to get connected users list", http.StatusInternalServerError)
+		return
+	}
+
+	b := &bytes.Buffer{}
+	csvWriter := csv.NewWriter(b)
+	if err := csvWriter.Write([]string{"Email"}); err != nil {
+		a.p.API.LogWarn("Unable to write headers in CSV file", "error", err.Error())
+		http.Error(w, "unable to write data in CSV file", http.StatusInternalServerError)
+		return
+	}
+
+	for _, email := range whitelist {
+		if err := csvWriter.Write([]string{email}); err != nil {
+			a.p.API.LogWarn("Unable to write data in CSV file", "error", err.Error())
+			http.Error(w, "unable to write data in CSV file", http.StatusInternalServerError)
+			return
+		}
+	}
+
+	csvWriter.Flush()
+	if err := csvWriter.Error(); err != nil {
+		a.p.API.LogWarn("Unable to flush the data in writer", "error", err.Error())
+		http.Error(w, "unable to write data in CSV file", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/csv")
+	w.Header().Set("Content-Disposition", "attachment;filename=invite-whitelist.csv")
+	if _, err := w.Write(b.Bytes()); err != nil {
+		a.p.API.LogWarn("Unable to write the data", "error", err.Error())
+		http.Error(w, "unable to write the data", http.StatusInternalServerError)
+	}
+}
+
+func (a *API) updateWhitelist(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		a.p.API.LogWarn("Not authorized")
+		http.Error(w, "not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	if !a.p.API.HasPermissionTo(userID, model.PermissionManageSystem) {
+		a.p.API.LogWarn("Insufficient permissions", "user_id", userID)
+		http.Error(w, "not able to authorize the user", http.StatusForbidden)
+		return
+	}
+
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		a.p.API.LogWarn("Error reading whitelist file")
+		http.Error(w, "error reading whitelist", http.StatusBadRequest)
+		return
+	}
+
+	reader := csv.NewReader(file)
+	columns, err := reader.Read()
+	if err != nil || strings.ToLower(columns[0]) != "email" || len(columns) != 1 {
+		a.p.API.LogWarn("Error parsing whitelist csv header")
+		http.Error(w, "error parsing whitelist - bad csv header", http.StatusBadRequest)
+		return
+	}
+
+	records, err := reader.ReadAll()
+	if err != nil {
+		a.p.API.LogWarn("Error parsing whitelist csv data")
+		http.Error(w, "error parsing whitelist", http.StatusBadRequest)
+	}
+
+	var users []*model.User
+
+	for _, row := range records {
+		email := row[0]
+
+		user, err := a.p.API.GetUserByEmail(email)
+		if err != nil {
+			a.p.API.LogWarn("Error getting user")
```

Comment:
```suggestion
			a.p.API.LogWarn("Error getting user", "email", email)
```

Reply:
Don't we want to avoid logging email (=sensitive data)

Reply:
Good point. In that case I'd suggest adding the `idx` to the `for ... range` loop and logging the line for the user not found. I'm basically trying to avoid a non-actionable logline here

Reply:
Indeed, I don't think we should be leaking email addresses. 2/5